client/src/featureform/serving.py

```diff
@@ -143,112 +84,211 @@ def _host_features(self, features, entities):
+            return all_feature_df
         except TypeError:
             print("Set is empty")
-        entity_row = all_feature_df.loc[all_feature_df[entity_tuple[0]] == entity_tuple[1]].copy()
-        entity_row.drop(columns=entity_tuple[0], inplace=True)
-        if len(entity_row.values) > 0:
-            return entity_row.values[0]
+
+    def get_features_for_entity(self, entity_id, entity_value, all_feature_df):
+        entity = all_feature_df.loc[all_feature_df[entity_id] == entity_value].copy()
+        entity.drop(columns=entity_id, inplace=True)
+        if len(entity.values) > 0:
+            return entity.values[0]
         else:
-            raise Exception("No matching entities for {}".format(entity_tuple))
+            raise Exception(f"No matching entities for {entity_id}: {entity_id}")
 
-    def process_feature_csv(self, source_path, entity_name, entity_col, value_col, dataframe_mapping,
-                            feature_name_variant, timestamp_column):
+    def feature_df_with_entity(self, source_path, entity_id, feature):
+        name_variant = f"{feature['name']}.{feature['variant']}"
         df = pd.read_csv(str(source_path))
-        if entity_col not in df.columns:
-            raise KeyError(f"Entity column does not exist: {entity_col}")
-        if value_col not in df.columns:
-            raise KeyError(f"Value column does not exist: {value_col}")
-        if timestamp_column != "" and timestamp_column not in df.columns:
-            raise KeyError(f"Timestamp column does not exist: {timestamp_column}")
-        if timestamp_column != "":
-            df = df[[entity_col, value_col, timestamp_column]]
-        else:
-            df = df[[entity_col, value_col]]
-        df.set_index(entity_col)
-        if timestamp_column != "":
-            df = df.sort_values(by=timestamp_column, ascending=True)
-        df.rename(columns={entity_col: entity_name, value_col: feature_name_variant}, inplace=True)
-        df.drop_duplicates(subset=[entity_name], keep="last", inplace=True)
-
-        if timestamp_column != "":
-            df = df.drop(columns=timestamp_column)
-        dataframe_mapping.append(df)
-        return dataframe_mapping
-
-    def process_label_csv(self, source_path, entity_name, entity_col, value_col, timestamp_column):
-        df = pd.read_csv(source_path)
-
-        if entity_col not in df.columns:
-            raise KeyError(f"Entity column does not exist: {entity_col}")
-        if value_col not in df.columns:
-            raise KeyError(f"Value column does not exist: {value_col}")
-        if timestamp_column != "" and timestamp_column not in df.columns:
-            raise KeyError(f"Timestamp column does not exist: {timestamp_column}")
-        if timestamp_column != "":
-            df = df[[entity_col, value_col, timestamp_column]]
-            df[timestamp_column] = pd.to_datetime(df[timestamp_column])
+        if feature['source_entity'] not in df.columns:
```

Comment:
We are using this check often. Do you think it might be better in it's own function?